packages/teuchos/comm/src/Teuchos_StackedTimer.cpp

```diff
@@ -600,6 +683,64 @@ StackedTimer::report(std::ostream &os, Teuchos::RCP<const Teuchos::Comm<int> > c
   }
 }
 
+void
+StackedTimer::reportXML(std::ostream &os, const std::string& datestamp, const std::string& timestamp, Teuchos::RCP<const Teuchos::Comm<int> > comm)
+{
+  flatten();
+  merge(comm);
+  OutputOptions defaultOptions;
+  collectRemoteData(comm, defaultOptions);
+  if (rank(*comm) == 0 ) {
+    std::vector<bool> printed(flat_names_.size(), false);
+    os << "<?xml version=\"1.0\"?>\n";
+    os << "<performance-report date=\"" << timestamp << "\" name=\"nightly_run_" << datestamp << "\" time-units=\"seconds\">\n";
+    printLevelXML("", 0, os, printed, 0.0);
+    os << "</performance-report>\n";
+  }
+}
+
+std::string
+StackedTimer::reportWatchrXML(const std::string& name, Teuchos::RCP<const Teuchos::Comm<int> > comm) {
+  char* rawWatchrDir = getenv("WATCHR_PERF_DIR");
```

Comment:
PLEASE cast this to `const char*`.  It's UB to modify the result, and we don't want to give other developers ideas ;-) .

Reply:
Good idea, done.